utils/port_utils.py

```
@@ -0,0 +1,259 @@
+
+
+def _acquire_file_lock() -> bool:
+    global _lock_fd
+    try:
+        import fcntl
+
+        _lock_fd = open(_LOCK_FILE, "w")
+        fcntl.flock(_lock_fd.fileno(), fcntl.LOCK_EX | fcntl.LOCK_NB)
+        _lock_fd.write(str(os.getpid()))
+        _lock_fd.flush()
+        return True
```

Ruff (0.15.1)

[warning] Consider moving this statement to an `else` block

(TRY300)